src/controllers/User.js

```diff
@@ -0,0 +1,170 @@
+const md5 = require("md5");
+const jwt = require("jsonwebtoken");
+const { reset_token } = require("../models/resetToken");
+const { User } = require("../models/Usermodel");
+const { access_token } = require("../models/Access_tokenModel");
+const { address } = require("../models/AddressModel");
+
+const Login = async (req, res) => {
+  try {
+    const user = await User.findOne({ username: req.body.username });
+    if (user) {
+      const givenPassword = md5(req.body.password);
+      if (user.password == givenPassword) {
+        const data = {
+          user_id: user._id,
+          token: jwt.sign(
+            {
+              user_id: user._id,
+            },
+            "secret",
+            { expiresIn: "1h" }
+          ),
+        };
+        await access_token.create(data);
+        res.send(data.token);
+      } else {
+        res.status(500).send("password not matched");
+      }
+    } else {
+      res.send("user not exists");
+    }
+  } catch (er) {
+    res.send(er);
+  }
+};
+const Register = async (req, res) => {
+  try {
+    if (req.body.password == req.body.conf_password) {
+      const data = {
+        username: req.body.username,
+        email: req.body.email,
+        firstname: req.body.firstname,
+        lastname: req.body.lastname,
+        password: md5(req.body.password),
+      };
+      await User.create(data);
+      res.send("usercreated");
+    } else {
+      res.send("password and confirm password did not matched");
+    }
+  } catch (er) {
+    res.send(er);
+  }
+};
+const UserDelete = async (req, res) => {
+  try {
+    const token = req.token;
+    const deletedUser = await User.deleteOne({ _id: token.user_id });
+    const deletionCount = await address.deleteMany({ user_id: token.user_id });
+    await access_token.deleteOne({ user_id: token.user_id });
```

Comment:
there is no condition that the access token can be create one time, 
but while deleting the token you are deleting only one, so fix this accordingly